src/Symfony/Component/Routing/Tests/Loader/AnnotationClassLoaderTest.php

```diff
@@ -11,245 +11,256 @@
+use Symfony\Component\Routing\Tests\Fixtures\AnnotationFixtures\LocalizedPrefixLocalizedActionController;
+use Symfony\Component\Routing\Tests\Fixtures\AnnotationFixtures\LocalizedPrefixMissingLocaleActionController;
+use Symfony\Component\Routing\Tests\Fixtures\AnnotationFixtures\LocalizedPrefixMissingRouteLocaleActionController;
+use Symfony\Component\Routing\Tests\Fixtures\AnnotationFixtures\LocalizedPrefixWithRouteWithoutLocale;
+use Symfony\Component\Routing\Tests\Fixtures\AnnotationFixtures\MethodActionControllers;
+use Symfony\Component\Routing\Tests\Fixtures\AnnotationFixtures\MissingRouteNameController;
+use Symfony\Component\Routing\Tests\Fixtures\AnnotationFixtures\NothingButNameController;
+use Symfony\Component\Routing\Tests\Fixtures\AnnotationFixtures\PrefixedActionLocalizedRouteController;
+use Symfony\Component\Routing\Tests\Fixtures\AnnotationFixtures\PrefixedActionPathController;
+use Symfony\Component\Routing\Tests\Fixtures\AnnotationFixtures\SymfonyRouteWithPrefixController;
+
+class AnnotationClassLoaderTest extends TestCase
 {
-    protected $loader;
-    private $reader;
+    /**
+     * @var AnnotationClassLoader
+     */
+    private $loader;
+
+    public function setUp()
+    {
+        $reader = new AnnotationReader();
+        $this->loader = new class($reader) extends AnnotationClassLoader {
+            protected function configureRoute(Route $route, \ReflectionClass $class, \ReflectionMethod $method, $annot) {}
+        };
+        AnnotationRegistry::registerLoader('class_exists');
+    }
+
+    /**
+     * @dataProvider provideTestSupportsChecksResource
+     */
+    public function testSupportsChecksResource($resource, $expectedSupports)
+    {
+        $this->assertSame($expectedSupports, $this->loader->supports($resource), '->supports() returns true if the resource is loadable');
+    }
 
-    protected function setUp()
+    public function provideTestSupportsChecksResource()
     {
-        parent::setUp();
+        return [
+            ['class', true],
+            ['\fully\qualified\class\name', true],
+            ['namespaced\class\without\leading\slash', true],
+            ['ÿClassWithLegalSpecialCharacters', true],
+            ['5', false],
+            ['foo.foo', false],
+            [null, false],
+        ];
+    }
 
-        $this->reader = $this->getReader();
-        $this->loader = $this->getClassLoader($this->reader);
+    public function testSupportsChecksTypeIfSpecified()
+    {
+        $this->assertTrue($this->loader->supports('class', 'annotation'), '->supports() checks the resource type if specified');
+        $this->assertFalse($this->loader->supports('class', 'foo'), '->supports() checks the resource type if specified');
     }
 
     /**
-     * @expectedException \InvalidArgumentException
+     * @test
      */
-    public function testLoadMissingClass()
+    public function simple_path_routes()
```

Comment:
You should not mix test method naming & follow Symfony ones already, so: `testSimplePathRoutes`.

Reply:
Corrected.